yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/identity/DigitalIdentityService.java

```diff
@@ -155,18 +196,18 @@ SignedRequest createSignedRequest(String sdkId, KeyPair keyPair, String path)
 
     SignedRequest createSignedRequest(String sdkId, KeyPair keyPair, String path, String method, byte[] payload)
             throws GeneralSecurityException, UnsupportedEncodingException, URISyntaxException {
-        SignedRequestBuilder requestBuilder = requestBuilderFactory.create()
+        SignedRequestBuilder request = requestBuilderFactory.create()
                 .withKeyPair(keyPair)
                 .withBaseUrl(apiUrl)
                 .withEndpoint(path)
                 .withHeader(AUTH_ID_HEADER, sdkId)
                 .withHttpMethod(method);
 
-        Optional.ofNullable(payload).map(v ->
-                requestBuilder.withPayload(v).withHeader(CONTENT_TYPE, CONTENT_TYPE_JSON)
-        );
+        if (null != payload) {
+            request.withPayload(payload).withHeader(CONTENT_TYPE, CONTENT_TYPE_JSON);
+        }
 
-        return requestBuilder.build();
+        return request.build();
```

Comment:
(only a suggestion)
```java
return Optional.ofNullable(payload)
        .map(request::withPayload)
        .map(r -> r.withHeader(..., ...))
        .orElse(request)
        .build()
```
If you rather keep the `if` I'd at least change it so it's more natural: 
```java
if (payload != null) {
    (...)
}
```